src/cpu/flexcpu/inflight_inst.hh

```diff
@@ -204,17 +261,28 @@ class InflightInst : public ExecContext,
     }
 
     inline bool isCommitted() const
-    { return status() == Committed; }
+    { return status() >= Committed; }
 
     inline bool isComplete() const
-    { return status() == Complete; }
+    { return status() >= Complete; }
+
+    inline bool isEffAddred() const
+    { return status() >= EffAddred; }
 
     inline bool isFaulted() const
     { return fault() != NoFault; }
 
+    inline bool isMemReady()
```

Comment:
What does this mean? If `isMemReady()` does that mean that you can send it to memory? Or...?

Reply:
Yes. Well, specifically that there are no longer any dependencies tracked by this InflightInst that would prevent it from going to memory. There are some minor nuances as to when it's sent to memory depending on how the dependencies are set up, but otherwise that is the core semantic.